actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -55,7 +55,11 @@ def unwrapped_exception
     end
 
     def rescue_template
-      @@rescue_templates[@exception.class.name]
+      @@rescue_templates.each do |error_class, template|
+        return template if error_class.safe_constantize === @exception
```

Comment:
This is a breaking change, isn't? Now the first exception that get matched will take precedence over others.

Reply:
It is, actually. I haven't looked at the currently registered errors, but the hierarchy check may introduce changes. Will take a look.